devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1602,6 +2821,40 @@ private def getStatusForDescription(changeType) {
 	return ""
 }
 
+/**
+ * Parse the time using SimpleDateFormat
+ *
+ * @param time: The time in HH:mm format
+ *
+ * @return The map with hour and minute
+ */
+private def getWeekDayHoursAndMinutes(time) {
+	def sdf = new SimpleDateFormat("HH:mm")
+	time = sdf.parse(time)
+	Calendar calendar = GregorianCalendar.getInstance()
+	calendar.setTime(time)
+	[hour: calendar.get(Calendar.HOUR_OF_DAY), minute: calendar.get(Calendar.MINUTE)]
+}
+
+/**
+ * Parse the date and time using SimpleDateFormat
+ *
+ * @param date: The date in yyyy-MM-dd format
+ *
+ * @param time: The time in HH:mm format
+ *
+ * @return The map with year, month, day, hour and minute
+ */
+private def getYearDayTimeDetails(date, time) {
+	def sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm")
```

Comment:
I might recommend doing the same splitting as above here as well on the date (using a hyphen), since you're just putting them into the `Calendar` and the immediately pulling them back out.